awssql/utils/sliding_expiration_cache.go

```diff
@@ -129,6 +128,19 @@ func (c *SlidingExpirationCache[T]) Remove(key string) {
 	}
 }
 
+func (c *SlidingExpirationCache[T]) cleanupIfExpired(key string) {
+	c.lock.Lock()
```

Comment:
Curious why the implementation is different from JDBC here?
JDBC also checks/updates `cleanupTimeNanos` time

Reply:
Note: the naming is a bit misleading, but Go's implementation of SlidingExpirationCache is really SlidngExpirationCacheWithCleanupThread. 

I was thinking it would be more efficient to only clean up the cache item in focus, and leave the other expired items to the clean up thread.

Do you have any thoughts/insights/suggestion on this?